src/environments/array.js

```diff
@@ -290,6 +299,10 @@ const htmlBuilder: HtmlBuilder<"array"> = function(group, options) {
         setHLinePos(hLinesBeforeRow[r + 1]);
     }
 
+    if (group.maxNumCols && nc > group.maxNumCols) {
+        throw new ParseError("Too many tab characters: &");
+    }
```

Comment:
This can be enforced in `parseArray` above, e.g.:
https://github.com/KaTeX/KaTeX/blob/e3c01376c85c82180f2985a18550e481f7e06b3f/src/environments/array.js#L123-L125
```js
        const token = parser.fetch();
        const next = token.text;
        if (next === "&") {
            if (group.maxNumCols && row.length > group.maxNumCols) {
                throw new ParseError("Too many tab characters: &", token);
            }
            parser.consume();
```

Reply:
Done.